lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -426,221 +350,134 @@ def get_user_narratives_ws_wsobjs(self, requesting_user, params, token):
+                                ws_nm = exec_task['job_input']['params'][0]['workspace_name']
+                                u_j_s['workspace_name'] = ws_nm
                     break
 
-            if not u_j_s.get('app_id') and u_j_s.get('method'):
+            if (not u_j_s.get('app_id') and u_j_s.get('method')):
                 u_j_s['app_id'] = u_j_s['method'].replace('.', '/')
 
+            if (not u_j_s.get('finish_time') and
+                        not u_j_s.get('error') and u_j_s.get('complete')):
+                u_j_s['finish_time'] = u_j_s.pop('modification_time')
+
             # get the narrative name and version if any
-            if u_j_s.get('wsid'):
+            if (u_j_s.get('wsid') and self.ws_narratives):
                 n_nm, n_obj = self.map_narrative(u_j_s['wsid'], self.ws_narratives)
                 if n_nm != "" and n_obj != 0:
                     u_j_s['narrative_name'] = n_nm
                     u_j_s['narrative_objNo'] = n_obj
 
-            # get some info from the client groups
+            # get the client groups
             u_j_s['client_groups'] = ['njs']  # default client groups to 'njs'
-            for clnt in self.client_groups:
-                clnt_app_id = clnt['app_id']
-                if (str(clnt_app_id).lower() == str(u_j_s.get('app_id')).lower()):
-                    u_j_s['client_groups'] = clnt['client_groups']
-                    break
-
-            # set the run/running/in_queue time
-            if u_j_s['job_state'] == 'completed' or u_j_s['job_state'] == 'suspend':
-                u_j_s['finish_time'] = u_j_s['modification_time']
-                u_j_s['run_time'] = u_j_s['finish_time'] - u_j_s['exec_start_time']
-            elif u_j_s['job_state'] == 'in-progress':
-                u_j_s['running_time'] = u_j_s['modification_time'] - u_j_s['exec_start_time']
-            elif u_j_s['job_state'] == 'queued':
-                u_j_s['time_in_queue'] = u_j_s['modification_time'] - u_j_s['creation_time']
+            if self.client_groups:
+                for clnt in self.client_groups:
+                    clnt_app_id = clnt['app_id']
+                    if (str(clnt_app_id).lower() == str(u_j_s.get('app_id')).lower()):
+                        u_j_s['client_groups'] = clnt['client_groups']
+                        break
 
             ujs_ret.append(u_j_s)
         return ujs_ret
 
-    def parse_app_id(self, lat):
+    def parse_app_id(self, exec_task):
         app_id = ''
-        if 'app_id' in lat['job_input']:
-            if '/' in lat['job_input']['app_id']:
-                app_id = lat['job_input']['app_id']
-            elif '.' in lat['job_input']['app_id']:
-                app_id = lat['job_input']['app_id'].replace('.', '/')
-            else:
-                app_id = lat['job_input']['app_id']
+        if 'app_id' in exec_task['job_input']:
+            app_id = exec_task['job_input']['app_id']
+            if '.' in app_id:
```

Comment:
You don't need this if statement. Just do `app_id = exec_task['job_input']['app_id'].replace('.', '/')`

Reply:
I was not sure if I'd get NoneType error, but I trust your expertise and will remove the unnecessary check.

Reply:
Okay, I just realize that you meant the inner 'if'.